src/Compilers/Core/Portable/Collections/TemporaryArray`1.cs

```diff
@@ -86,6 +87,17 @@ private TemporaryArray(in TemporaryArray<T> array)
 #pragma warning restore RS0042 // Do not copy value
         }
 
+        public static TemporaryArray<T> GetInstance(int capacity)
+        {
+            if (capacity <= InlineCapacity)
+                return Empty;
```

Comment:
capacity less than this amount is already supported by the Empty array value.  so can just return that immediately.